gcc/rust/rust-session-manager.cc

```diff
@@ -330,6 +328,9 @@ Session::init ()
   // setup backend to GCC GIMPLE
   backend = rust_get_backend ();
 
+  // set the default crate name
+  options.set_crate_name (kDefaultCrateName);
```

Comment:
The default in rustc is the source filename without `.rs` or `rust_out` if the source is stdin.

Reply:
Lets raise an issue to fix that default crate name, I think it makes sense to fix that outside of this PR

Reply:
Another question: what if the source doesn't have a `.rs` extension?

Reply:
https://github.com/rust-lang/rust/blob/758c00ea4088e1ca2714890f00c791c24f001536/compiler/rustc_session/src/output.rs#L48

Rustc first tries `--crate-name` and `#![crate_name]` (if both are used they must match), then tries to take the filename, take the file stem (filename with last extension removed if any. `a.tar.gz` would result in `a.tar` for the file stem, which is not a valid crate name) and replace `-` with `_`. And finally falls back to `rust_out` if the input is not a file. The file extension doesn't matter.

Reply:
We still need to figure out how we can handle ```#![crate_name]``` as we don't have a way to feed this back to the compiler driver yet to auto set the output name.

Reply:
See #789.